lib/dgram.js

```diff
@@ -272,6 +267,65 @@ Socket.prototype.send = function(buffer,
   if (offset + length > buffer.length)
     throw new RangeError('Offset + length beyond buffer length');
 
+  return buffer.slice(offset, offset + length);
+}
+
+function fixBuffer(buffer) {
+  var length = 0;
+
+  for (var i = 0, l = buffer.length; i < l; i++) {
+    if (typeof buffer[i] === 'string')
+      buffer[i] = new Buffer(buffer[i]);
```

Comment:
Agreed, let's finalize this and then I can submit another PR to handle the strings. However, there is the encoding issue to be solved.